### PR TITLE
[System] Remove unused BTLS P/Invokes.

### PR DESCRIPTION
**Pull request by rolfbjarne**
They point to internal functions that don't exist, causing compilation
problems for Xamarin.Mac apps.

Example test/compile failure: https://gist.github.com/rolfbjarne/43cf2228816bbe8a863f331da3e07c0d

### CODE REVIEW

File: mcs/class/System/mobile_System.dll.sources [hunk 1]
```diff
@@ -305,7 +305,6 @@ Mono.Btls/MonoBtlsX509LookupMonoCollection.cs
 Mono.Btls/MonoBtlsX509LookupMono.cs
 Mono.Btls/MonoBtlsX509LookupType.cs
 Mono.Btls/MonoBtlsX509Name.cs
-Mono.Btls/MonoBtlsX509NameList.cs
```
Comment (akoeplinger) [inline mcs/class/System/mobile_System.dll.sources]: don't remove this from the System sources, it'd break monodroid. The files have #ifdefs in them so they're only compiled when BTLS is enabled.

Comment (rolfbjarne) [inline mcs/class/System/mobile_System.dll.sources]: Why do you want to compile a file that's contains code that's never used?

**Comment by rolfbjarne** [comment]
@akoeplinger The point here is that all the removed code is dead code for everybody, not just Xamarin.Mac.

And this happens when Xamarin.Mac references the system Mono's dlls, which are not compiled with any XAMMAC defines.

File: mcs/class/System/mobile_System.dll.sources (hunk 1 above)
Comment (akoeplinger) [inline mcs/class/System/mobile_System.dll.sources]: My assumption was that this file is there for a reason, that's why Martin should take a look. If it's really dead code then we should of course remove it :)

**Comment by akoeplinger** [comment]
> And this happens when Xamarin.Mac references the system Mono's dlls, which are not compiled with any XAMMAC defines.

Right, I forgot about this scenario.

**Comment by baulig** [comment]
In general, we should not remove the actual sources as several things that may not be used at the moment are actually there for a reason and will be used later.  However, this particular file is a left-over from a previous source cleanup, so it can safely be deleted.

**Pull request merged**